src/FakeItEasy/Core/FakeManager.EventRule.cs

```diff
@@ -191,7 +184,7 @@ private EventCall(EventInfo @event, MethodInfo callingMethod, Delegate eventHand
                         return null;
                     }
 
-                    var eventInfos = eventAdderOrRemover.DeclaringType.GetEvents(
+                    var eventInfos = eventAdderOrRemover.DeclaringType!.GetEvents(

```

Comment:
DeclaringType can only be null for non-nested types, module-level methods, or dynamic methods.